src/fastapibootcamp/handlers/external.py

```diff
@@ -317,3 +330,135 @@ async def post_log_demo(
         greeting=greeting_templates[data.language].format(name=data.name),
         language=data.language,
     )
+
+
+# =============================================================================
+# Lesson 5: Handling internal errors
+#
+# Sometimes things go wrong in your application. A database doesn't respond,
+# an external service is down, or some bug in your code causes an exception.
+# By default, FastAPI returns a 500 Internal Server Error response when an
+# uncaught exception occurs. Safir provides the ability to report these
+# uncaught errors to a Slack channel through an incoming webhook.
+#
+# To hook up this Slack reporting, you need to take the following steps
+# 1. Add a slack_webhook_url field to the configuration (config.py)
+# 2. In main.py, configure the SlackRouteErrorHandler middleware
+# 3. Create custom SlackException subclasses for your application (usually
+#    in exceptions.py)
+# 4. Raise these exceptions in your application code
+# 5. Optionally, add SlackRouteErrorHandler to the APIRouter in your
+#    handlers/endpoints module to automatically report any uncaught exception
+#    to Slack.
```

Comment:
This isn't entirely optional in the sense that if you don't do this, your raised exceptions won't be reported to Slack. (Although you can explicitly initialize a Safir Slack client and manually send exceptions to Slack that way.)